client/v2/common/common.go

```diff
@@ -103,27 +103,27 @@ func mergeRawQueries(q1, q2 string) string {
 }
 
 // submitFormRaw is a helper used for submitting (ex.) GETs and POSTs to the server
-func (client *Client) submitFormRaw(ctx context.Context, path string, body interface{}, requestMethod string, encodeJSON bool, headers []*Header) (resp *http.Response, err error) {
+func (client *Client) submitFormRaw(ctx context.Context, path string, body interface{}, requestMethod string, encodeJSON bool, headers []*Header, getParams interface{}) (resp *http.Response, err error) {
```

Comment:
What do you think about renaming body to params, and having the new parameter be body?

That way you don't have this weird conditional on the requestMethod to decide how to handle parameters.